src/osx/core/strconv_cf.cpp

```diff
@@ -89,27 +89,23 @@ WXDLLIMPEXP_BASE wxMBConv* new_wxMBConv_cf(wxFontEncoding encoding)
         if ( theString == NULL )
             return wxCONV_FAILED;
 
+        if ( m_compose_decompose )
+        {
         // Ensure that the string is in canonical composed form (NFC): this is
         // important because Darwin uses decomposed form (NFD) for e.g. file
         // names but we want to use NFC internally.
         wxCFRef<CFMutableStringRef>
             cfMutableString(CFStringCreateMutableCopy(NULL, 0, theString));
         CFStringNormalize(cfMutableString, kCFStringNormalizationFormC);
         theString = cfMutableString;
-
+        }
+        
         /* NOTE: The string content includes the NULL element if the source string did
          * That means we have to do nothing special because the destination will have
          * the NULL element iff the source did and the NULL element will be included
          * in the count iff it was included in the source count.
          */
 
-
```

Comment:
These simplifications are definitely very nice, it just would have been great to have them in a separate commit to make things more clear...